taskana-adapter-camunda-listener/src/main/java/pro/taskana/adapter/camunda/parselistener/TaskanaParseListenerProcessEnginePlugin.java

```diff
@@ -35,60 +39,95 @@ public void preInit(ProcessEngineConfigurationImpl processEngineConfiguration) {
     initOutbox(processEngineConfiguration);
   }
 
-  public static DataSource createDatasource(
-      String driver, String jdbcUrl, String username, String password) {
-    return new PooledDataSource(driver, jdbcUrl, username, password);
-  }
-
   private void initParseListeners(ProcessEngineConfigurationImpl processEngineConfiguration) {
 
-    List<BpmnParseListener> preParseListeners =
-        processEngineConfiguration.getCustomPreBPMNParseListeners();
+    try {
 
-    if (preParseListeners == null) {
-      preParseListeners = new ArrayList<BpmnParseListener>();
-      processEngineConfiguration.setCustomPreBPMNParseListeners(preParseListeners);
-    }
+      List<BpmnParseListener> preParseListeners =
+          processEngineConfiguration.getCustomPreBPMNParseListeners();
 
-    preParseListeners.add(new TaskanaParseListener());
+      if (preParseListeners == null) {
+        preParseListeners = new ArrayList<>();
+        processEngineConfiguration.setCustomPreBPMNParseListeners(preParseListeners);
+      }
+      preParseListeners.add(new TaskanaParseListener());
+
+      LOGGER.debug("TaskanaParseListener registered successfully");
+
+    } catch (Exception e) {
 
-    LOGGER.debug("TaskanaParseListener registered successfully");
+      LOGGER.warn("Caught exception while trying to register TaskanaParseListener", e);
+
+      throw new SystemException(
+          "An error occured while trying to register the TaskanaParseListener."
+              + " Aborting the boot of camunda.");
+    }
   }
 
   private void initOutbox(ProcessEngineConfigurationImpl processEngineConfiguration) {
-    boolean processEngineMustBeClosed = false;
-    DataSource dataSource = processEngineConfiguration.getDataSource();
-    if (dataSource == null) {
-      LOGGER.warn(
-          "ProcessEngineConfiguration returns null DataSource. "
-              + "Retrieving DataSource from properties.");
-      dataSource = getDataSourceFromPropertiesFile();
-      if (dataSource == null) {
+
+    DataSource camundaDataSource = retrieveCamundaDatasource(processEngineConfiguration);
+
+    createSchema(camundaDataSource);
+
+    LOGGER.info("TaskanaOutbox initialized successfully");
+  }
+
+  private void createSchema(DataSource camundaDataSource) {
+
+    String outboxSchema = initSchemaName(camundaDataSource);
+
+    TaskanaOutboxSchemaCreator schemaCreator =
+        new TaskanaOutboxSchemaCreator(camundaDataSource, outboxSchema);
+
+    boolean schemaIsPreexisting = schemaCreator.isSchemaPreexisting();
+
+    boolean createSchema = isAutomatedSchemaCreationEnabled();
```

Comment:
there are too many things called "createSchema": the current method, this boolean flag and the method of SchemaCreator.
Proposal: start boolean varables with an 'is' or 'should'. So this variable would read 'isSchemaToBeCreated' or 'shouldSchemaBeCreated'. The other variable would be 'isSchemaPreexisting'